packages/atomic/src/components/search/atomic-results-per-page/atomic-results-per-page.ts

```diff
@@ -0,0 +1,149 @@
+import {
+  buildResultsPerPage,
+  buildSearchStatus,
+  type ResultsPerPage,
+  type ResultsPerPageState,
+  type SearchStatus,
+  type SearchStatusState,
+} from '@coveo/headless';
+import {html, LitElement} from 'lit';
+import {customElement, property, state} from 'lit/decorators.js';
+import {renderFieldsetGroup} from '@/src/components/common/fieldset-group.js';
+import {createAppLoadedListener} from '@/src/components/common/interface/store.js';
+import {renderChoices} from '@/src/components/common/items-per-page/choices.js';
+import {
+  ChoiceIsNaNError,
+  InitialChoiceNotInChoicesError,
+} from '@/src/components/common/items-per-page/error.js';
+import {renderLabel} from '@/src/components/common/items-per-page/label.js';
+import {
+  convertChoicesToNumbers,
+  validateInitialChoice,
+} from '@/src/components/common/items-per-page/validate.js';
+import {renderPagerGuard} from '@/src/components/common/pager/pager-guard.js';
+import type {Bindings} from '@/src/components/search/atomic-search-interface/interfaces.js';
+import {bindStateToController} from '@/src/decorators/bind-state.js';
+import {bindingGuard} from '@/src/decorators/binding-guard.js';
+import {errorGuard} from '@/src/decorators/error-guard.js';
+import type {InitializableComponent} from '@/src/decorators/types.js';
+import {withTailwindStyles} from '@/src/decorators/with-tailwind-styles.js';
+import {InitializeBindingsMixin} from '@/src/mixins/bindings-mixin.js';
+import {randomID} from '@/src/utils/utils.js';
+
+/**
+ * The `atomic-results-per-page` component determines how many results to display per page.
+ *
+ * @part label - The "Results per page" label.
+ * @part buttons - The list of buttons.
+ * @part button - The result per page button.
+ * @part active-button - The active result per page button.
+ */
+@customElement('atomic-results-per-page')
+@withTailwindStyles
```

Comment:
```suggestion
@bindings()
@withTailwindStyles
```

You can have both the decorator and the mixin, once we migrate the interface the mixin can be removed on all the components for that use case